core/src/main/java/org/elasticsearch/action/delete/DeleteResponse.java

```diff
@@ -104,15 +85,10 @@ public static void parseXContentFields(XContentParser parser, Builder context) t
      */
     public static class Builder extends DocWriteResponse.Builder {
 
-        private boolean found = false;
-
-        public void setFound(boolean found) {
-            this.found = found;
-        }
-
         @Override
         public DeleteResponse build() {
-            DeleteResponse deleteResponse = new DeleteResponse(shardId, type, id, seqNo, primaryTerm, version, found);
+            DeleteResponse deleteResponse = new DeleteResponse(shardId, type, id, seqNo, primaryTerm, version,
```

Comment:
I think you should pass the Result here. Maybe also add an assertion in the ctor that the result is either `Result.DELETED` or `Result.NOT_FOUND`.